peer.go

```diff
@@ -473,87 +473,82 @@ func (p *peer) readNextMessage() (lnwire.Message, error) {
 	return nextMsg, nil
 }
 
-// chanMsgStream implements a goroutine-safe, in-order stream of messages to be
-// delivered to an active channel. These messages MUST be in order due to the
-// nature of the lightning channel commitment state machine. We utilize
-// additional synchronization with the fundingManager to ensure we don't
-// attempt to dispatch a message to a channel before it is fully active.
-type chanMsgStream struct {
-	fundingMgr *fundingManager
-	htlcSwitch *htlcswitch.Switch
+// msgStream implements a goroutine-safe, in-order stream of messages to be
+// delivered via closure to a receiver. These messages MUST be in order due to
+// the nature of the lightning channel commitment and gossiper state machines.
+type msgStream struct {
+	peer *peer
 
-	cid lnwire.ChannelID
+	apply func(lnwire.Message)
```

Comment:
For the sake of logging, I think we should also add an `error` return parameter. It may help us track down issues int he future with misbehaving sub-systems.

Reply:
left off the returned error for now, as the closure may have more context on how to print the err message, plus we don't really handle the primary errors from inside the closure